src/Barberry/Plugin/Imagemagick/Converter.php

```diff
@@ -25,11 +27,21 @@ public function configure(ContentType $targetContentType, $tempPath)
     public function convert($bin, Plugin\InterfaceCommand $command = null)
     {
         $shellCommand = new ShellCommand($command);
+
+        $matches = [];
+        preg_match_all('@\[(\d+)\]\sGIF\s(\d+x\d+)\s(.*?)\s(.*?)\s(.*?)\s@s', $bin, $matches);
+        if (isset($matches[1], $matches[2]) && !empty($matches[1]) && count($matches[1]) > 1) {
+            $shellCommand = new AnimatedGif($command, new Meta($matches));
+        }
```

Comment:
Content type recognition is responsibility of Libmagic.

Reply:
done

Reply:
Imagemagick is too heavy dependency, please, before going further, answer the first question.

Background: barberry already has content type recognition in its core, plugins shouldn't duplicate this functionality.

Reply:
I understand, but here I should know what command I need, since to convert animated gif properly I have to prepare CLI command in a bit different way

Reply:
So, what is the difference processing animated GIFs?

Reply:
you have to provide a geometry width and height of image (in context of animated gif - frame's geometry)

Reply:
Let's find a way to make your solution work, maybe you'll be able to invent a command suitable for any GIF image? You would be able to create a dedicated plugin for such images.